meshroom/nodes/general/InputFile.py

```diff
@@ -0,0 +1,55 @@
+__version__ = "1.0"
+
+import logging
+import os
+
+from meshroom.core import desc
+
+logger = logging.getLogger(__name__)
+
+class InputFile(desc.InputNode, desc.InitNode):
+    """
+This node is an input node that receives a File.
+"""
+    category = "Other"
+
+    inputs = [
+        desc.File(
+            name="inputFile",
+            label="Input File",
+            description="A file or folder to use as the input.",
+            value="",
+        )
+    ]
+
+    def initialize(self, node, inputs, recursiveInputs):
+        logger.setLevel(logging.INFO)
+        self.resetAttributes(node, ["inputFile"])
+
+        attributeSet = False
+        if len(inputs) >= 1:
+            for i in inputs:
+                if os.path.isfile(i) or os.path.isdir(i):
+                    self.setAttributes(node, {"inputFile": i})
+                    attributeSet = True
+
+                    if len(inputs) > 1:
+                        logger.info(f"Several inputs were provided ({inputs}).")
+                        logger.info(f"Only the first one ({i}) will be used.")
+
+                    break
+
+        if not attributeSet and len(recursiveInputs) >= 1:
```

Comment:
no need for "attributeSet", only need an if/elif/else.

Reply:
I would prefer to keep it simple/strict and have explicit errors.

Reply:
Done